File: java/savesys.java
// Savesys, an easily-implementable save manager
// Written by draumaz in 2021
// BSD 3-Clause

import java.io.File;
import java.io.FileNotFoundException;
import java.io.PrintWriter;
import java.util.Arrays;
import java.io.FileWriter;
import java.io.BufferedWriter;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.util.List;

class savesys {
    public static String save_name() {
        return "temp.txt"; // default filename {CONFIG}
    }
    public static int save_default() {
        return 10; // default line variables {CONFIG}
    }
    public static void exists() {
        if (new File(save_name()).exists());
        else
            generate();
    }
    public static int generate() {
        try {
                new File(save_name()).createNewFile(); // generate file
            } catch (IOException I) {
                return 1; 
            }
            try {
                BufferedWriter buf = null;
                buf = new BufferedWriter(new FileWriter(save_name())); // open writer
                for (int fr = 0; fr < save_default(); fr++) { 
                    buf.write(0+"\n"); // loop write save_default() variables
                }
                buf.flush();
                buf.close();
            } catch (IOException io) {
                return 1;
            }
            return 0;
    }
    public static int[] read() {
		int[] wr = new int[save_default()]; // set array size
		try {
			List<String> z = Files.readAllLines(Paths.get(save_name())); // pull into list
			String[] str_arr = z.stream().toArray(String[]::new); // convert to string array
			for (int fr = 0; fr < wr.length; fr++) {
				wr[fr] = Integer.parseInt(str_arr[fr]); // loop convert into int array
			}
		} catch (IOException io) {}
		  catch (ArrayIndexOutOfBoundsException a) {
		  	generate();
			System.out.println("\nSave corrupted. Continuing with a new save.");
		}
		return wr;
	}
    public static int write(int line, int state) {
		int[] r = read(); // get current save
		try {
			BufferedWriter buf = null;
			buf = new BufferedWriter(new FileWriter(save_name())); // initalize writer
			for (int i = 0; i < r.length; i++) {
                if (i == line) {
                    buf.write(Math.round((int)(state < 0? 0 : state))+"\n"); // write requested state to line
                    continue;
                }
				buf.write(r[i]+"\n"); // loop write to file
			}
			buf.flush();
			buf.close();
		} catch (IOException io) {
            return 1;
        }
        return 0;
    }
}

File: py/savesys.py
# Savesys, an easily-implementable save manager
# draumaz, 2021 | GPL v3

import os

## begin config ##
save_name = "default.txt"
save_size = 20
save_contents = "0"
## end config ##

def gen():
    open(save_name, "w+").write(save_size*(save_contents+"\n"))
    
def exists():
    if not (os.path.exists(save_name)): # If the file doesn't exist...
        gen() # ...create it, and fill it using config values.

def reader():
    return list(map(int, open(save_name, "r").readlines())) # Return the entire file as an array of integers.

def writer(line, state):
    arr = read() # Retrieve the current states...
    f = open(save_name, "w") # ...open the file...
    for i in range(0, save_size): # ...set loop ranges....
        if i == line:
            f.write(str(state)+"\n") # ...if the loop matches line, write state...
            continue
        f.write(str(arr[i])+"\n") # ...and write to the file!
